packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -888,8 +904,30 @@ private void destroyMapViewIfNecessary() {
     if (mapView == null) {
       return;
     }
-    mapView.onDestroy();
+
+    final MapView mapReference = mapView; // keep a reference to the mapView for the callback
     mapView = null;
+
+    // This fixes an issue where the mapView is still being used by the render thread after disposal.
+    // Delaying the actual mapView disposal to the next frame avoids the issue.
+    Runnable r =
+        () -> {
+          Log.e(TAG, "Mapview destroy in callback");
```

Comment:
Please remove the error logging for things that aren't errors.

Reply:
Yep, those were some debugging leftovers, should be gone now.